fiber/sentryfiber.go

```diff
@@ -0,0 +1,120 @@
+package sentryfiber
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
+
+	"github.com/getsentry/sentry-go"
+)
+
+const valuesKey = "sentry"
+
+type handler struct {
+	repanic         bool
+	waitForDelivery bool
+	timeout         time.Duration
+}
+
+type Options struct {
+	// Repanic configures whether Sentry should repanic after recovery, in most cases it should be set to false,
+	// as fasthttp doesn't include it's own Recovery handler.
+	Repanic bool
+	// WaitForDelivery configures whether you want to block the request before moving forward with the response.
+	// Because fasthttp doesn't include it's own Recovery handler, it will restart the application,
+	// and event won't be delivered otherwise.
+	WaitForDelivery bool
+	// Timeout for the event delivery requests.
+	Timeout time.Duration
+}
+
+func New(options Options) fiber.Handler {
+	handler := handler{
+		repanic:         options.Repanic,
+		timeout:         time.Second * 2,
+		waitForDelivery: options.WaitForDelivery,
+	}
+
+	if options.Timeout != 0 {
+		handler.timeout = options.Timeout
+	}
+
+	return handler.handle
+}
+
+func (h *handler) handle(ctx *fiber.Ctx) error {
+	hub := sentry.CurrentHub().Clone()
+	scope := hub.Scope()
+	scope.SetRequest(convert(ctx))
+	scope.SetRequestBody(ctx.Request().Body())
+	ctx.Locals(valuesKey, hub)
+	defer h.recoverWithSentry(hub, ctx)
+	return ctx.Next()
+}
```

Comment:
Should we support automatic HTTP server spans creation right now? Or do you plan to have it on a separate PR?

Reply:
I can address that in a separate PR (and add for all remaining integrations as well).